src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -65,6 +69,102 @@ static char *IsolationLevelName[] = {
 };
 
 
+PG_FUNCTION_INFO_V1(citus_internal_start_management_transaction);
+PG_FUNCTION_INFO_V1(execute_command_on_other_nodes);
+PG_FUNCTION_INFO_V1(citus_mark_object_distributed);
+PG_FUNCTION_INFO_V1(commit_management_command_2PC);
+
+bool IsManagementCommand = false;
+FullTransactionId outerXid;
+static MultiConnection *managementCon = NULL;
+bool CitusManagementDatabase = true;
+
+Datum
+citus_internal_start_management_transaction(PG_FUNCTION_ARGS)
+{
+	outerXid = PG_GETARG_FULLTRANSACTIONID(0);
+	IsManagementCommand = true;
+
+	PG_RETURN_VOID();
+}
+
+
+Datum
+execute_command_on_other_nodes(PG_FUNCTION_ARGS)
+{
+	text *queryText = PG_GETARG_TEXT_P(0);
+	char *query = text_to_cstring(queryText);
+
+	List *queryList = NIL;
+	queryList = lappend(queryList, "SET citus.enable_create_role_propagation TO OFF;");
+	queryList = lappend(queryList, query);
+	queryList = lappend(queryList, "SET citus.enable_create_role_propagation TO ON;");
+	List *taskList = NodeDDLTaskList(NON_COORDINATOR_NODES, queryList);
```

Comment:
Given the name of the function, it would be good to introduce an OTHER_NODES option, such that it also works correctly from worker nodes.